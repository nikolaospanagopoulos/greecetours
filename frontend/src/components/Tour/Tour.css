.card{
    background-image: linear-gradient(to bottom right, #ffb366, #ff8080);
    border-radius: 20px;
    box-shadow: 0 0 30px rgba(0,0,0,0.18);
    padding: 0.1rem;
    width: 25rem;
    overflow: hidden;

}



.container{
    position: relative;
    clip-path: polygon(0 0,100% 0,100% 92%, 0 100%);
}

.card-image{
    
    width: 100%;
    height: 18em;
    display: block;
    border-radius: 20px 20px 0px 0;
}





@media screen and (max-width: 768px){
    .card{
        width: 90vw;
        
    }
    .card-image{
        height: 55vh;
    }
}

