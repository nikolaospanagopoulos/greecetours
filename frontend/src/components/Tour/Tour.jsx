import React from 'react';
import './Tour.css';
import {Link} from 'react-router-dom'
import Rating from '../Rating/Rating';


const Tour = ({tour}) => {
    return ( 
        <div className='card'>
            <div className='container'>
            <Link to={`/tours/tour/${tour._id}`}>
                <img src={tour.image1} alt="tour poster" className='card-image'/>
            </Link>
            </div>
            
            <div className='details'>
            <Link to={`/tours/tour/${tour._id}`}>
                <h3> {tour.name} </h3>
                </Link>
                <Rating value={tour.rating} text={`${tour.numReviews} reviews`}/>
                <p>
                â‚¬{tour.price}
                </p>
            </div>
        </div>
     );
}
 
export default Tour;