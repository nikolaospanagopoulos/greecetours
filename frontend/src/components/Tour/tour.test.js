
import { shallow } from "enzyme";
import React from 'react'
import Tour from "./Tour";


it('expect to render Tour Component',() => {
    const tour = [
        {
            
            tour:'Athens'
        }
    ]
    expect(shallow(<Tour tour={tour}/> ).length).toEqual(1)
})

it('take a tour snapshot',() => {
    const tour = [
        {
            
            tour:'Athens'
        }
    ]
    expect(shallow(<Tour  tour={tour}/>)).toMatchSnapshot()
})

