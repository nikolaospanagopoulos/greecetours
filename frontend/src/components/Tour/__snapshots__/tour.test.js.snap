// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`take a tour snapshot 1`] = `
<div
  className="card"
>
  <div
    className="container"
  >
    <Link
      to="/tours/tour/undefined"
    >
      <img
        alt="tour poster"
        className="card-image"
      />
    </Link>
  </div>
  <div
    className="details"
  >
    <Link
      className="product-link"
      to="/tours/tour/undefined"
    >
      <h3>
         
         
      </h3>
    </Link>
    <Rating
      color="blue"
      numreviews="undefined reviews"
    />
    <p>
      â‚¬
    </p>
  </div>
</div>
`;
